plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -2999,9 +3000,21 @@ protected void saveTransOrJob( ISharedObjectsTransformer objectTransformer, Repo
           new RepositoryFile.Builder( file ).title( RepositoryFile.DEFAULT_LOCALE, element.getName() ).createdDate(
               versionDate != null ? versionDate.getTime() : new Date() ).description( RepositoryFile.DEFAULT_LOCALE,
               Const.NVL( element.getDescription(), "" ) ).build();
-      file =
+      try {
```

Comment:
Need to see a test case before this can be merged.

Reply:
Added reproduce scenario.